src/Lecture3.hs

```diff
@@ -125,10 +134,15 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+    (<>) :: Reward -> Reward -> Reward
+    x <> y = case (rewardSpecial x, rewardSpecial y) of
+        (True, _) -> Reward {rewardGold=rewardGold x <> rewardGold y, rewardSpecial=True}
+        (_, True) -> Reward {rewardGold=rewardGold x <> rewardGold y, rewardSpecial=True}
+        (_, _) -> Reward {rewardGold=rewardGold x <> rewardGold y, rewardSpecial=False}
 
 instance Monoid Reward where
-
+    mempty :: Reward
+    mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
    mempty = Reward mempty False
```